experimental/devices/mfrc522/rf522.go

```diff
@@ -0,0 +1,763 @@
+// - auth - the authentiction mode.
+// - sector - the sector on the card to write to.
+// - block - the block within the sector to write into.
+// - data - 16 bytes if data to write
+// - key - the key used to authenticate the card - depends on the used auth method.
+func (r *Dev) WriteBlock(auth byte, sector int, block int, data [16]byte, key []byte) (err error) {
+	defer func() {
+		r.StopCrypto()
+	}()
+	uuid, err := r.selectCard()
+	if err != nil {
+		return
+	}
+	state, err := r.Auth(auth, sector, 3, key, uuid)
+	if err != nil {
+		return
+	}
+	if state != AuthOk {
+		err = errors.New("Authentication failed")
+		return
+	}
+
+	err = r.write(calcBlockAddress(sector, block%3), data[:])
+	return
+}
+
+// ReadSectorTrail reads the sector trail (the last sector that contains the sector access bits)
+// - sector - the sector number to read the data from.
+func (r *Dev) ReadSectorTrail(sector int) (res []byte, err error) {
+	res, err = r.read(calcBlockAddress(sector&0xFF, 3))
+	return
+}
+
+// WriteSectorTrail writes the sector trait with sector access bits.
+// - auth - authentication mode.
+// - sector - sector to set authentication.
+// - keyA - the key used for AuthA authentication scheme.
+// - keyB - the key used for AuthB authentication schemd.
+// - access - the block access structure.
+// - key - the current key used to authenticate the provided sector.
+func (r *Dev) WriteSectorTrail(auth byte, sector int, keyA [6]byte, keyB [6]byte, access *BlocksAccess, key []byte) (err error) {
+	defer func() {
+		r.StopCrypto()
+	}()
+	uuid, err := r.selectCard()
+	if err != nil {
+		return
+	}
+	state, err := r.Auth(auth, sector, 3, key, uuid)
+	if err != nil {
+		return
+	}
+	if state != AuthOk {
+		err = errors.New("Failed to authenticate")
+		return
+	}
+
+	data := make([]byte, 16)
+	copy(data, keyA[:])
+	accessData := CalculateBlockAccess(access)
+	copy(data[6:], accessData[:4])
+	copy(data[10:], keyB[:])
+	err = r.write(calcBlockAddress(sector&0xFF, 3), data)
+	return
+}
+
+// Auth authenticate the card fof the sector/block using the provided data.
+// - mode - the authentication mode.
+// - sector - the sector to authenticate on.
+// - block - the block within sector to authenticate.
+// - sectorKey - the key to be used for accessing the sector data.
+// - serial - the serial of the card.
+func (r *Dev) Auth(mode byte, sector, block int, sectorKey []byte, serial []byte) (authS AuthStatus, err error) {
+	authS, err = r.auth(mode, calcBlockAddress(sector, block), sectorKey, serial)
+	return
+}
+
+func (r *Dev) selectCard() (uuid []byte, err error) {
+	err = r.Wait()
+	if err != nil {
+		return
+	}
+	err = r.Init()
+	if err != nil {
+		return
+	}
+	_, err = r.Request()
+	if err != nil {
+		return
+	}
+	uuid, err = r.AntiColl()
+	if err != nil {
+		return
+	}
+	_, err = r.SelectTag(uuid)
+	if err != nil {
+		return
+	}
+	return
+}
+
+// ReadCard reads the card sector/block.
+// - auth - the authentication mode.
```

Comment:
Always keep an empty line to correctly format the documentation.

https://blog.golang.org/godoc-documenting-go-code is a good introduction.